### DISCUSSION TITLE
3.0.8r - FLAC cause Guru Meditation Error: Core  1 panic'ed

### DISCUSSION THREAD

**Discussion by Marcin-Fenger** [schreibfaul1/ESP32-audioI2S#704] [General]
Hi
Please take a look at FLAC decoding in recent realase.

[116835][I][Audio.cpp:5375] ts_parsePacket(): parseTS reset
Connect to new host: "http://192.168.1.47:2222/amplituner/Europe.flac"
buffers freed, free Heap: 156444 bytes
connect to 192.168.1.47 on port 2222 path /amplituner/Europe.flac
Connection has been established in 11 ms, free Heap: 155848 bytes
[117196][I][Audio.cpp:4047] parseContentType(): ContentType audio/x-flac, format is flac
[117201][I][Audio.cpp:3881] parseHttpResponseHeader(): Switch to DATA, metaint is 0
content-length: 60540731
FLACDecoder has been initialized, free Heap: 142316 bytes , free stack 14968 DWORDs
http://192.168.1.47:2222/amplituner/Europe.flac
stream ready, buffer filled in 1143 ms
FLAC maxBlockSize: 1152
FLAC maxFrameSize: 3942
FLAC sampleRate: 44100
FLAC numChannels: 2
FLAC bitsPerSample: 16
total samples in stream: 25623864
audio file duration: 581 seconds
id3data
TITLE: Europe Endless
id3data
ALBUM: Trans-Europe Express
id3data
TRACKNUMBER: 1
id3data
ARTIST: Kraftwerk
id3data
COMMENT: FLAC settings: exhaustiveModelSearch:1 midSideSteGuru Meditation Error: Core  1 panic'ed (InstrFetchProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x00000000  PS      : 0x00060b30  A0      : 0x800ebfc7  A1      : 0x3ffe9320
A2      : 0x3ffc4218  A3      : 0x3ffc42b0  A4      : 0x3ffc4ce0  A5      : 0x00000000  
A6      : 0x00000064  A7      : 0x00000001  A8      : 0x800e59df  A9      : 0x3ffe9300
A10     : 0x3ffc42b0  A11     : 0x000003d8  A12     : 0x00001fec  A13     : 0x00000c40  
A14     : 0x007bf768  A15     : 0x003fffff  SAR     : 0x0000000a  EXCCAUSE: 0x00000014
EXCVADDR: 0x00000000  LBEG    : 0x40086a25  LEND    : 0x40086a2d  LCOUNT  : 0x00000027  

Backtrace: 0xfffffffd:0x3ffe9320 0x400ebfc4:0x3ffe93b0 0x400f2c21:0x3ffe93e0 0x400f39bd:0x3ffe9420 0x400d6205:0x3ffe9440

  #0  0xfffffffd:0x3ffe9320 in ?? ??:0
  #1  0x400ebfc4:0x3ffe93b0 in Audio::read_FLAC_Header(unsigned char*, unsigned int) at lib/ESP32-audioI2S-master/src/Audio.cpp:1450
      (inlined by) Audio::readAudioHeader(unsigned int) at lib/ESP32-audioI2S-master/src/Audio.cpp:1282
  #2  0x400f2c21:0x3ffe93e0 in Audio::processWebFile() at lib/ESP32-audioI2S-master/src/Audio.cpp:3319
  #3  0x400f39bd:0x3ffe9420 in Audio::loop() at lib/ESP32-audioI2S-master/src/Audio.cpp:2378 (discriminator 1)     
  #4  0x400d6205:0x3ffe9440 in Task2Audio(void*) at src/main.cpp:3024

ELF file SHA256: 2f7b7c9eefab86f3

Other formats works well. On 3.0.0 also works.

CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp-wrover-kit.html
PLATFORM: Espressif 32 (6.5.0) > Espressif ESP-WROVER-KIT
HARDWARE: ESP32 240MHz, 320KB RAM, 16MB Flash //n16r8
DEBUG: Current (ftdi) On-board (ftdi) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20014.231204 (2.0.14)
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5

**Comment by schreibfaul1** [comment]
Can you please check again if the URL is correct, 192.168.1.47 does not return a ping, it looks more like a home network.

**Reply by Marcin-Fenger** [reply]
It is local network. There are some mp3 files too and work well. It's not network issue.

**Reply by Marcin-Fenger** [reply]
Together with my firend Piotr We just have done some tests and I have got additional info. Flac radio streams works. Flac files without jpg works. Flac with jpg inside hungs the system. We try it on two independed systems and reason is the same.
